src/Data/Aeson/Extended.hs

```diff
@@ -150,7 +150,7 @@ instance IsString WarningParserMonoid where
 
 -- Parsed JSON value with its warnings
 data WithJSONWarnings a = WithJSONWarnings a [JSONWarning]
-    deriving Generic
+    deriving (Eq, Generic, Show)
```

Comment:
Is this change necessary for the PR, or just to help with debugging?

Reply:
`Eq` and `Show` are required for test code like:

```haskell
decode' contents `shouldBe` Just (WithJSONWarnings expected [])
```